source/extensions/filters/http/header_to_metadata/header_to_metadata_filter.cc

```diff
@@ -50,7 +60,15 @@ bool Config::configToVector(const ProtobufRepeatedRule& proto_rules,
       throw EnvoyException(error);
     }
 
-    vector.push_back(rule);
+    // Ensure value and regex_value_rewrite are not mixed.
+    if (entry.has_on_header_present()) {
+      auto& on_header_present = entry.on_header_present();
+      if (!on_header_present.value().empty() && on_header_present.has_regex_value_rewrite()) {
```

Comment:
same here, you can pull out `on_header_present` regardless, since the default value will pass the validation